File: C++/74. Search a 2D Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();
        int n = matrix[0].size();
        int l = 0;
        int r = m*n-1;
        while (l<=r){
            int mid = (l+r)/2;
            int i=mid/n;
            int j=mid%n; 
            if (matrix[i][j]==target){
                return true;
            }
            else if(matrix [i][j]<target){
                l=mid + 1;
            }
            else{
                r=mid-1;
            }
        }
        return false;
    }
};

File: JavaScript/74. Search a 2D Matrix.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    var m = matrix.length;
    var n = matrix[0].length;
    var l = 0;
    var r = m*n-1;
    while (l<=r){
        var mid = Math.floor((l+r)/2);
        var i=Math.floor(mid/n);
        var j=Math.floor(mid%n); 
        if (matrix[i][j]==target){
            return true;
        }
        else if(matrix [i][j]<target){
            l=mid + 1;
        }
        else{
            r=mid-1;
        }
    }
    return false;
};
